ENIGMAsystem/SHELL/Platforms/General/PFmain.cpp

```diff
@@ -23,6 +25,27 @@ int frames_count = 0;
 unsigned long current_time_mcs = 0;
 bool game_window_focused = true;
 
+void game_format_caption() {
+  std::ostringstream out;
+  // round ALL the numbers, like 5.5 to 6 lives
+  // e.g, GM8.1 behavior
+  out.precision(0);
+
+  // DON'T append anything that isn't set so the main loop
+  // could optimize the caption out completely
+  if (!((std::string)enigma_user::room_caption).empty())
+    out << (std::string)enigma_user::room_caption << " ";
+  if (enigma_user::show_score && enigma_user::score != 0)
+    out << enigma_user::caption_score << std::fixed << enigma_user::score << " ";
+  if (enigma_user::show_lives)
+    out << enigma_user::caption_lives << std::fixed << (double)enigma_user::lives << " ";
+  if (enigma_user::show_health)
+    out << enigma_user::caption_health << std::fixed << enigma_user::health;
+
+  std::string caption = out.str();
+  if (!caption.empty()) enigma_user::window_set_caption(caption);
```

Comment:
I just now realized that this means that if the `room_caption` wasn't empty, and then it became empty, the window caption would not actually be updated to an empty string. I am not really sure if that's what we intended when we originally added this check.

Though it's unlikely anybody will run into this bug because they'll just set the room caption empty to begin with and then use `window_set_caption` most likely. If they did run into this, they could just work around it by also calling `window_set_caption` after they set the `room_caption` back to nothing. This could be more problematic considering the score though, which, when 0, isn't supposed to be visible.